mask/eolearn/mask/masking.py

```diff
@@ -11,24 +11,32 @@
 This source code is licensed under the MIT license found in the LICENSE
 file in the root directory of this source tree.
 """
-from typing import Callable, Union
+from __future__ import annotations
+
+from typing import Callable, List, Union
 
 import numpy as np
 
-from eolearn.core import EOTask, ZipFeatureTask
+from eolearn.core import EOPatch, EOTask, FeatureType, ZipFeatureTask
+from eolearn.core.types import FeaturesSpecification, Literal, SingleFeatureSpec
 
 
 class JoinMasksTask(ZipFeatureTask):
     """Joins together masks with the provided logical operation."""
 
-    def __init__(self, input_features, output_feature, join_operation: Union[str, Callable] = "and"):
+    def __init__(
+        self,
+        input_features: FeaturesSpecification,
+        output_feature: SingleFeatureSpec,
+        join_operation: Union[Literal["and", "or", "xor"], Callable] = "and",
+    ):
         """
         :param input_features: Mask features to be joined together.
         :param output_feature: Feature to which to save the joined mask.
         :param join_operation: How to join masks. Supports `'and'`, `'or'`, `'xor'`, or a `Callable` object.
         """
         input_features = self.parse_features(input_features)
-        output_feature = self.parse_feature(output_feature)
+        output_feature = self.parse_feature(output_feature)  # type: ignore[assignment]
```

Comment:
these are not used in the init anyway, let the super-class parse them

Reply:
done